lib/src/json_schema/json_schema.dart

```diff
@@ -523,7 +523,8 @@ class JsonSchema {
           }
 
           // If currentSchema is a ref, resolve ref recursively.
-          if (currentSchema.ref != null) {
+          // Cycle detection happens at evaluation time for drafts 2019 and 2020.
+          if (currentSchema.schemaVersion < SchemaVersion.draft2019_09 && currentSchema.ref != null) {
```

Comment:
This language is used in the draft7 spec:

> A schema MUST NOT be run into an infinite loop against a schema.  For
   example, if two schemas "#alice" and "#bob" both have an "allOf"
   property that refers to the other, a naive validator might get stuck
   in an infinite recursive loop trying to validate the instance.
   Schemas SHOULD NOT make use of infinite recursive nesting like this;
   the behavior is undefined.

So I don't think we want this additional check